### PR TITLE
Add e2e tests for onboarding process

### LINKED ISSUE #1286: Create e2e tests for the Create New Wallet onboarding flow
## Detailed Description

Create e2e tests for the Create New Wallet onboarding flow

## Context

We need a way to test this onboarding flow as part of our CI process

## Possible Implementation

Implement Spectron tests that:

 - Start the app
 - Select the Create New Wallet flow
 - Fill out the forms
 - Verify that the process completes from start to finish without any unexpected errors

### LINKED ISSUE #1287: Create e2e tests for the Import Existing Wallet onboarding flow
## Detailed Description

Create e2e tests for the Import Existing Wallet onboarding flow

## Context

We need a way to test this onboarding flow as part of our CI process

## Possible Implementation

Implement Spectron tests that:

 - Start the app
 - Select the Import Existing Wallet flow
 - Fill out the forms
 - Verify that the process completes from start to finish without any unexpected errors

### LINKED ISSUE #1288: Create e2e tests for the Connect Your Own Node onboarding flow
## Detailed Description

Create e2e tests for the Connect Your Own Node onboarding flow

## Context

We need a way to test this onboarding flow as part of our CI process

## Possible Implementation

Implement Spectron tests that:

 - Start the app
 - Select the Connect Your Own Node flow
 - Fill out the forms
 - Verify that the process comp

We will either need to spin up an lnd node on the CI server that we can connect to, or host one externally.

### LINKED ISSUE #1289: Create e2e tests for the BTCPay Server onboarding flow
## Detailed Description

Create e2e tests for the BTCPay Server onboarding flow

## Context

We need a way to test this onboarding flow as part of our CI process

## Possible Implementation

Implement Spectron tests that:

 - Start the app
 - Select the BTCPay Server flow
 - Fill out the forms
 - Verify that the process comp

We will either need to spin up an lnd node on the CI server that we can connect to, or host one externally.

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

- migrate e2e test suite to testcafe
- write basic tests to cover onboarding process

## Motivation and Context:

- Close #1286 
- Close #1287
- Close #1288
- Close #1289

## How Has This Been Tested?

`yarn test-e2e`

## Types of changes:

 - New test framework
 - New tests

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [x] I have added tests to cover my changes where needed.
- [x] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: app/lib/zap/controller.js [hunk 1]
```diff
@@ -149,20 +151,16 @@ class ZapController {
...
+    if (this.walletUnlocker && this.walletUnlocker.can('disconnect')) {
+      this.walletUnlocker.disconnect()
     }
+    this.sendMessage('lndStopped')
+
+    // Shut down Neutrino.
+    await this.shutdownNeutrino()
 
     // Give the grpc connections a chance to be properly closed out.
     await new Promise(resolve => setTimeout(resolve, 200))
```
Comment (korhaliv) [inline app/lib/zap/controller.js]: consider using 
`https://github.com/LN-Zap/zap-desktop/blob/master/app/lib/utils/promiseTimeout.js`

File: app/lib/zap/controller.js [hunk 2]
```diff
@@ -419,7 +426,7 @@ class ZapController {
         this.neutrino.removeListener('exit', exitHandler)
         if (this.neutrino) {
           mainLog.warn('Graceful shutdown failed to complete within 10 seconds.')
-          this.neutrino.kill('SIGTERM')
+          this.neutrino.kill('SIGKILL')
```
Comment (korhaliv) [inline app/lib/zap/controller.js]: Consider extracting a constant since it's also used down below

File: app/lib/zap/controller.js (hunk 1 above)
Comment (mrfelton) [reply app/lib/zap/controller.js]: I'll raise a separate PR for refactoring that.

File: app/lib/zap/controller.js (hunk 2 above)
Comment (mrfelton) [reply app/lib/zap/controller.js]: Where it's used below the usage is different.

Here, we are passing the `SIGKILL` signal. Where as below we are not passing a specific signal and using the default value of `SIGINT` which is defined in the `Neutrino.prototype.kill` method.

I'm not sure that there is any useful consolidation that can be done here given the usage of each call is deliberately different.

Comment (mrfelton) [reply app/lib/zap/controller.js]: Just realised that you are referring to the `SIGKILL` value, not the method call!

Comment (korhaliv) [reply app/lib/zap/controller.js]: Yes I'm referring to this:
[https://github.com/LN-Zap/zap-desktop/blob/92b21d9e09b0c1c90fce5d41fd0cb4af269c1293/app/lib/zap/controller.js#L498](https://github.com/LN-Zap/zap-desktop/blob/92b21d9e09b0c1c90fce5d41fd0cb4af269c1293/app/lib/zap/controller.js#L498)

Comment (mrfelton) [reply app/lib/zap/controller.js]: Although - I'm still not sure of the benefit of making this a constant in this instance...

Comment (korhaliv) [reply app/lib/zap/controller.js]: Maybe. I just usually extract if i have > 1 usages. And in general don't like string literals.

Comment (mrfelton) [reply app/lib/zap/controller.js]: I tend to agree if these are string that we are defining for our own purposes. But `SIGINT` and `SIGTERM` etc are standard process signals that can be passed to external processes. These are defined by node itself https://nodejs.org/api/process.html#process_signal_events

**Pull request merged**